### PR TITLE
[Form] fix group sequence based validation

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20929
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php [hunk 1]
```diff
@@ -349,13 +348,27 @@ function () { throw new TransformationFailedException(); }
...
+    {
+        $object = $this->getMock('\stdClass');
         $options = array('validation_groups' => array($this, 'getValidationGroups'));
         $form = $this->getBuilder('name', '\stdClass', $options)
             ->setData($object)
             ->getForm();
 
-        $this->expectValidateAt(0, 'data', $object, 'group1');
-        $this->expectValidateAt(1, 'data', $object, 'group2');
+        $this->expectValidateAt(0, 'data', $object, array('group1', 'group2'));
```
Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php]: Looks like this bug fix is breaking BC with this behavioral change. But it seems the code was broken so I'm still 👍 for this one.

**Review by HeahDude — APPROVED** [review]

File: src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php]: @HeahDude Which BC break do you refer to?

Comment (HeahDude) [inline src/Symfony/Component/Form/Tests/Extension/Validator/Constraints/FormValidatorTest.php]: Arff, got confused by your refactoring :)

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**